Upgrade kotlin-maven-plugin

dessalines opened the pull request:
Had a compile error without this upgrade.

dessalines committed:
Upgrade kotlin-maven-plugin

mesmere commented on pom.xml:
```diff
@@ -30,7 +30,7 @@
                 <groupId>org.jetbrains.kotlin</groupId>
                 <artifactId>kotlin-maven-plugin</artifactId>
-                <version>1.4.21</version>
+                <version>1.6.20</version>
```
Hm, if I update this then the kotlin libraries in the dependencies should be updated too.

mesmere committed:
Specify explicit kotlin.version in pom

This should fix broken builds in specifically newer versions of intellij 

(maven on the command line worked fine afaict)

rileynull commented:
Thank you!
